tt_eager/tt_dnn/op_library/bmm/kernels/dataflow/reader_bmm_tile_layout_in1_sender_writer_padding.cpp

```diff
@@ -213,6 +269,32 @@ void kernel_main() {
                 uint64_t in3_start_address = l1_write_addr_in3; // copy start address of block, to be used for mcasting
                 uint32_t in3_block_size_bytes = 0; // can be optimized later, pass it to kernel
 
+                #ifdef IN1_DRAM_SHRADED
+                uint32_t l1_write_addr_in3_offset = 0;
+                uint32_t index_offset = 0;
+
+                for (uint32_t i = 0; i < num_dram_shards_to_read; ++i) {
+                    uint32_t in3_base_addr = noc_async_read_tile_dram_sharded_set_state<bias_single_tile_size_bytes, false>(in3_tensor_addr, current_dram_bank_id[index_offset]);
+
+                    if (i == 0) {
+                        in3_base_addr += dram_tensor_start_offset;
+                    }
+
+                    uint32_t l1_read_addr_in3 = 0;
+                    uint32_t l1_write_addr_in3 = get_write_ptr(cb_id_in3) + l1_write_addr_in3_offset;
+                    uint32_t in3_block_w_dram = in1_block_w_dram_stride_bytes[index_offset] / bias_single_tile_size_bytes;
+
+                    for (uint32_t w = 0; w < in3_block_w_dram; ++w) {
+                        noc_async_read_tile_dram_sharded_with_state(in3_base_addr, l1_read_addr_in3, l1_write_addr_in3);
+                        l1_read_addr_in3 += bias_single_tile_size_bytes;
+                        l1_write_addr_in3 += bias_single_tile_size_bytes;
+                        in3_block_size_bytes += bias_single_tile_size_bytes;
+                    }
+                    l1_write_addr_in3_offset += in1_block_w_dram_stride_bytes[index_offset];
+                    index_offset += 2;
```

Comment:
Please make the 2 a variable with a descriptive name. There's not enough context to know why you're adding by 2.

Reply:
You're also adding by 2 on line 205 above.

Reply:
changed to next_bank_id_and_dram_stride_index